test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1

```diff
@@ -22,28 +35,119 @@ Describe "Job Cmdlet Tests" -Tag "CI" {
             $waitjob = Wait-Job -Timeout 10 -id $j.id
             if ( $waitJob ) {
                 $result = receive-job -id $j.id
-                $result.id | Should be 2
+                $result | Should be 2
             }
         }
     }
     Context "jobs which take time" {
+        BeforeAll {
+            $jobBlock =
+            {
+                $count = 5
+
+                while ($count)
+                {
+                    Write-Output "count: $count"
+                    --$count
+                    start-sleep 2
+                }
+            }
+        }
         BeforeEach {
-            $j = start-job -scriptblock { Start-Sleep 15 }
+            $j = start-job -scriptblock $jobBlock
         }
         AfterEach {
             Get-Job | Remove-Job -force
         }
         It "Wait-Job will wait for a job" {
             $result = wait-Job $j
             $result | should be $j
+            $j.State | should be Completed
         }
-        It "Stop-Job will stop a job" -pending {
+        It "Wait-Job will timeout waiting for a job" {
+            $result = Wait-Job -Timeout 2 $j
+            $result | should be $null
+        }
+        It "Stop-Job will stop a job" {
             Stop-Job -id $j.id
-            $j.Status |Should be Stopped
+            $j.State | should be Stopped
+        }
+        It "Remove-Job will not remove a running job" {
+            $id = $j.Id
+            $j | remove-job -ErrorAction SilentlyContinue
+            $job = Get-Job -Id $id
+            $job | should be $j
+        }
+        It "Remove-Job -Force will remove a running job" {
+            $id = $j.Id
+            $j | remove-job -ErrorAction SilentlyContinue
+            $job = Get-Job -Id $id
+            $job | should be $j
+            $j | Remove-Job -Force
+            $job = Get-Job -Id $id -ErrorAction SilentlyContinue
+            $job | should be $null
+        }
+    }
+    Context "Retrieving partial output from jobs" {
+        BeforeAll {
+            function getResults($job, $n, $keep)
+            {
+                $results = @()
+
+                while ($results.Count -lt $n)
+                {
+                    if ($keep)
+                    {
+                        $results = @() + (Receive-Job -Keep $job)
+                    }
+                    else
+                    {
+                        $results += Receive-Job $job
+                    }
+                }
+
+                return $results
+            }
+
+            # call this only within an It block
+            function checkContent($ar)
+            {
+                $v = $ar[0] - 1
+
+                for ($i=0; $i -lt $ar.Count; ++$i)
+                {
+                    $ar[$i] | should be ($v + 1)
+                    $v = $ar[$i]
+                }
+            }
+
+        }
+        BeforeEach {
+            $j = start-job -scriptblock { $count = 1; while ($true) { Write-Output ($count++) } }
```

Comment:
Perhaps it is justified in these tests.
If no I would prefer synchronous and more deterministic tests.
0.25 is 250 ms so we could use file IO to be even faster.
1. The test write a test string to a file.
2. "start-job" job read (Get-Content -wait) the string and send to output
3. The test call Receive-Job (in loop up to 5 sec) to get the test string